src/tests/ftest/soak/soak.py

```diff
@@ -120,65 +119,192 @@ def remote_copy(self, hostlist, remote_dir, local_dir):
+                for b_size in bsize_list:
+                    for t_size in tsize_list:
+                        for o_type in oclass_list:
+                            ior_cmd = IorCommand()
+                            ior_cmd.namespace = ior_params
+                            ior_cmd.get_params(self)
+                            if iteration is not None and iteration < 0:
+                                ior_cmd.repetitions.update(1000000)
+                            if self.job_timeout is not None:
+                                ior_cmd.max_duration.update(self.job_timeout)
+                            else:
+                                ior_cmd.max_duration.update(10)
+                            ior_cmd.api.update(api)
+                            ior_cmd.block_size.update(b_size)
+                            ior_cmd.transfer_size.update(t_size)
+                            ior_cmd.daos_oclass.update(o_type)
+                            ior_cmd.set_daos_params(self.server_group, pool)
+
+                            # slurm/sbatch cmdline
+                            if self.job_manager == "Sbatch":
+                                # export the user environment for srun
+                                exports = ["ALL"]
+                                if ior_cmd.api.value == "MPIIO":
+                                    env = {
+                                        "CRT_ATTACH_INFO_PATH": os.path.join(
+                                            self.basepath, "install/tmp"),
+                                        "DAOS_POOL": str(
+                                            ior_cmd.daos_pool.value),
+                                        "DAOS_CONT": str(
+                                            ior_cmd.daos_cont.value),
+                                        "MPI_LIB": "\"\"",
+                                        "DAOS_SVCL": str(
+                                            ior_cmd.daos_svcl.value),
+                                        "DAOS_SINGLETON_CLI": 1,
+                                        "FI_PSM2_DISCONNECT": 1,
+                                        "IOR_HINT__MPI__romio_daos_obj_class":
+                                        str(ior_cmd.daos_oclass.value)
+                                    }
+                                    exports.extend(["{}={}".format(
+                                        key, val) for key, val in env.items()])
+                                cmd = "srun -l --mpi=pmi2 --export={} {}".format(
+                                    ",".join(exports), str(ior_cmd))
+                                commands.append([job_spec, cmd, task])
+                            else:
+                                commands.append([job_spec, ior_cmd, task])
+                            self.log.info("<<IOR cmdline>>: %s \n",
+                                          commands[-1][1].__str__())
+        return commands
+
+    def create_dmg_cmdline(self, job_spec, pool):
```

Comment:
(pylint-unused-argument) Unused argument 'job_spec'
(pylint-unused-argument) Unused argument 'pool'
(pylint-no-self-use) Method could be a function